### PR TITLE
Use ruff instead of pylint and black

### PR DESCRIPTION
**Pull request by FoamyGuy**
I got setup locally to test another PR on this repo so I've gone ahead and done the ruff conversion to submit here.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# SPDX-FileCopyrightText: 2024 Justin Myers for Adafruit Industries
+#
+# SPDX-License-Identifier: Unlicense
+
+* text eol=lf
```
Comment (tannewt) [inline .gitattributes:5]: Same as the other PR. I think this is too broad and should be limited to some file extensions in this repo.

**Review by tannewt — CHANGES_REQUESTED** [review]
One gitattributes comment. Good otherwise. Thank you!

File: .gitattributes (hunk 1 above)
Comment (FoamyGuy) [reply .gitattributes]: The latest commit changes this to use a more specific list of extensions.

**Review by tannewt — APPROVED** [review]
Thank you!

**Pull request merged**